lib/cocoapods-core/lockfile.rb

```diff
@@ -92,9 +92,21 @@ def version(pod_name)
       pod_versions[root_name]
     end
 
+    # Returns the source of the given Pod.
+    #
+    # @param [String] pod_name The name of the Pod (root name of the specification).
+    #
+    # @return [String] The source of the pod.
+    #
+    # @return [Nil] If there is no source stored for the given name.
+    #
+    def spec_repo(pod_name)
+      pods_by_spec_repo.detect { |key, values| break key if values.include?(pod_name) }
+    end
```

Comment:
Should this be cached/memoized in an internal hash? Since https://github.com/CocoaPods/CocoaPods/pull/7464 will use this in a loop, it will end up traversing several times

Reply:
dont think its that much of a problem. The maximum number of times `spec_repo` method can be called is N where N is the number of pods you have and it means that every single one changed its source for this method to be invoked.

I think N ranges from 1-200 here.